dev/com.ibm.ws.microprofile.health.2.0/src/com/ibm/ws/microprofile/health20/internal/HealthCheck20HttpResponseBuilder.java

```diff
@@ -67,9 +66,23 @@ protected void setChecks(HealthCheckResponse response) {
             overallState = State.DOWN; // treat as fail case
         }
 
-        Optional<Map<String, Object>> data = response.getData();
-        if ((data != null) && data.isPresent())
-            check.put(HealthCheckConstants.HEALTH_CHECK_PAYLOAD_DATA, data.get());
+        Optional<Map<String, Object>> dataMap = response.getData();
+        if ((dataMap != null) && dataMap.isPresent()) {
+            JsonObjectBuilder data = jsonBuilderFactory.createObjectBuilder();
+            for (Map.Entry<String, Object> entry : dataMap.get().entrySet()) {
+                Object value = entry.getValue();
+                if (value == null) {
+                    data.add(entry.getKey(), JsonValue.NULL);
+                } else if (value instanceof String) {
+                    data.add(entry.getKey(), (String) value);
+                } else if (value instanceof Boolean) {
+                    data.add(entry.getKey(), (Boolean) value);
+                } else {
+                    data.add(entry.getKey(), (Long) value);
```

Comment:
Is it guaranteed to be a Long or do we need to do an instanceof?

Reply:
As far as I can tell from HealthCheckResponseBuilderImpl, it will be a String, Boolean, or Long.